### PR TITLE
fix: support localized lens and billing pages

### LINKED ISSUE #185: [bug] lens list/get fails: .lens_edit_lens wrapper removed and aria-labels localized
### What happened?

`kagi lens list` and `kagi lens get` fail with a parse error on accounts whose Kagi UI language is not English (reproduced with Norwegian):

```
$ kagi lens list
parse error: lens '4248' missing edit url
```

### Root cause

`parse_lens_list` in `src/parser.rs` resolves the edit link with:

```rust
let edit_selector = selector(r#".lens_edit_lens a[aria-label="Edit lens"]"#)?;
```

Two problems with the live page (verified against `/html/settings/lenses` server HTML, 2026-09-12):

1. **The `.lens_edit_lens` wrapper no longer exists.** The edit anchor now sits directly inside the `form.__lens_item`:
```html
<a aria-label='Rediger linse'
    class="btn-icon m-n4"
    href="/settings/update_lens?id=4248">…svg…</a>
```
2. **`aria-label` is localized** — a Norwegian-locale account renders `aria-label='Rediger linse'`, so even unwrapping the selector would not match non-English accounts.

This means every lens fails at the `edit_url` extraction, aborting the whole list.

### Suggested fix

Fall back to matching the edit link by `href` prefix, which is locale-independent and wrapper-independent:

```rust
let edit_fallback_selector = selector(r#"a[href^="/settings/update_lens?id="]"#)?;
// …
let edit_url = item
    .select(&edit_selector)
    .next()
    .or_else(|| item.select(&edit_fallback_selector).next())
    .and_then(|node| node.value().attr("href"))
    // …
```

I have implemented and tested this locally: `lens list` returns all 14 lenses and `lens get` works, E2E against a Norwegian-locale account. Happy to open a PR with the fix + a regression test using the live markup above — just say the word (or I can attach the full patch here).

### Environment

- kagi-cli 0.20.0 (homebrew tap `microck/kagi`)
- macOS 15 (arm64)
- Kagi UI language: Norwegian (Bokmål)

### LINKED ISSUE #186: [bug] usage hard-fails on Starter plan (no AI-cost box) and localized labels
### What happened?

`kagi usage` fails hard on plans that don't include the AI-cost box (reproduced on Starter):

```
$ kagi usage
parse error: Kagi billing page did not contain the current AI cost and limit; the page layout may have changed
```

### Root cause

`parse_usage_html` in `src/usage.rs` requires the label `total ai cost this period` (or `ai usage (usd)`) and returns a parse error when absent. But the AI-cost box is an Ultimate-plan feature. A Starter billing page (verified against `/settings/billing` server HTML, 2026-09-12, Norwegian locale) instead renders plain usage counters:

```html
<span>Starter</span> …
<div class="billing_box_count_title">Søk</div>
<div class="billing_box_count_num">…3…/300</div>
<div class="billing_box_count_title">Assistant-interaksjoner</div>
<div class="billing_box_count_num"><span >8</span>/300</div>
<div class="billing_box_count_title">Saldo</div>
<div class="billing_box_count_num">$0.07</div>
<span>Neste fornyelse er <b>2026-09-19</b></span>
```

Two separate issues:

1. **`ai_cost` is treated as mandatory**, so every non-Ultimate account gets a hard parse error even though plan/balance/renewal data is present.
2. **Labels are matched in English only.** `account balance` / `next renewal is` never match localized pages (`Saldo`, `Neste fornyelse er`), so those fields come back empty on non-English accounts even when present.

### Suggested fix

- Make `ai_cost: Option<AiCostUsage>` and skip the field when the box is absent (JSON consumers already tolerate `skip_serializing_if`-style fields).
- Accept localized variants for the labels that have stable equivalents (or, better long-term, parse the `billing_box_count_*` structure instead of label text).

I have implemented this locally (optional `ai_cost` + `saldo`/`neste fornyelse er` fallbacks) and verified E2E on a Norwegian-locale Starter account:

```json
{
  "plan": "Starter",
  "ai_cost": null,
  "account_balance_usd": 0.07,
  "next_renewal": "2026-09-19",
  "daily_usage": []
}
```

All 215 unit tests pass (one test updated: the "missing cost pair errors" test now asserts successful parse with `ai_cost: None`). Happy to open a PR.

### Environment

- kagi-cli 0.20.0 (homebrew tap `microck/kagi`)
- macOS 15 (arm64)
- Kagi plan: Starter; UI language: Norwegian (Bokmål)

### PR DESCRIPTION
**Pull request by Microck**
## Summary

localized Kagi accounts broke every lens command that needs the edit URL, and Starter plan accounts could not use `kagi usage` at all.

this reads lens edit links from their stable route, treats the plan-specific AI cost box as optional, and recognizes the Norwegian balance and renewal labels reported in the issues.

Closes #185.
Closes #186.

## Verification

- [x] `cargo fmt --check`
- [x] `cargo clippy --all-targets --all-features -- -D warnings`
- [x] `cargo test --workspace --all-targets --locked` (317 passed, 13 credential-gated tests ignored)
- [x] localized lens parser regression test
- [x] localized Starter usage parser regression test, including `"ai_cost": null`
- [x] `pnpm build` from `docs/` (44 static pages)
- [x] `(cd npm && npm pack --dry-run)`

## Docs

- [x] usage reference updated
- [x] changelog updated
- [x] README does not need a change

## Auth / secrets

- [x] no tokens, cookies, or local config secrets committed
- [x] no live authenticated requests were made by the agent

## AI assistance

- `agent_name`: OpenAI Codex
- `agent_version`: codex-cli 0.154.0
- `model_used`: GPT-5
- `human_testing`: issue reporter Terje Pettersen manually tested `lens list` and `lens get` on a Norwegian-locale account as documented in #185, and tested `usage` on a Norwegian-locale Starter account as documented in #186. The local checks listed above were run by OpenAI Codex, not a human.
- `contribution_summary`: fixed localized lens edit-link parsing and billing usage parsing for plans without AI cost tracking.

## Summary by CodeRabbit

- **Bug Fixes**
  - Lens listing and related commands now recognize edit links across account languages.
  - `kagi usage` now supports plans without AI cost tracking.
  - Usage reports now correctly read balances and renewal dates from localized billing pages.
  - AI cost details are shown only when available.

- **Documentation**
  - Updated `kagi usage` documentation to explain when AI cost fields appear in JSON responses.

### CODE REVIEW

**Comment by Microck** [comment]
> [!NOTE]
> 🤖 GPT-5 responding on behalf of Microck

this warning treats the abbreviated HTML excerpt in #186 as the full page. the excerpt contains `…` after `<span>Starter</span>`. the reporter's tested patch only makes `ai_cost` optional and adds the localized balance and renewal labels, and their human E2E output already contains `"plan": "Starter"`.

the existing plan parser is therefore working against the full live page. adding a fallback for any standalone `Starter` span would guess from unrelated page text. the regression fixture intentionally covers only the parser behavior changed here.

File: src/usage.rs [hunk 1]
```diff
@@ -154,25 +154,22 @@ pub fn format_pretty(report: &UsageReport) -> String {
...
     let candidates = text_candidates(&document);
-    let ai_cost = extract_ai_cost(&candidates).ok_or_else(|| {
-        KagiError::Parse(
-            "Kagi billing page did not contain the current AI cost and limit; the page layout may have changed"
-                .to_string(),
-        )
-    })?;
 
-    Ok(UsageReport {
+    UsageReport {
```
[the remark this answers is not in the document]
Comment (Microck) [reply src/usage.rs]: > [!NOTE]
> 🤖 GPT-5 responding on behalf of Microck

fixed in `f1e8451`. the parser now requires at least one billing-specific value while still allowing the AI cost box to be absent. a new test confirms unrelated or changed HTML returns a parse error.

File: src/usage.rs [hunk 2]
```diff
@@ -154,25 +154,22 @@ pub fn format_pretty(report: &UsageReport) -> String {
...
         plan: extract_plan(&candidates),
-        ai_cost,
-        account_balance_usd: extract_first_decimal_after_label(&candidates, ACCOUNT_BALANCE_LABEL),
-        next_renewal: extract_text_after_label(&candidates, NEXT_RENEWAL_LABEL)
+        ai_cost: extract_ai_cost(&candidates),
```
[the remark this answers is not in the document]
Comment (Microck) [reply src/usage.rs]: > [!NOTE]
> 🤖 GPT-5 responding on behalf of Microck

the issue excerpt is abbreviated with `…` after the Starter span. the reporter's human-tested patch did not change plan parsing, and their full live response already contains `"plan": "Starter"`. this fixture covers the markup relevant to the change, not the omitted plan-price markup. matching an arbitrary standalone `Starter` span would add an unverified parser path, so I am leaving this unchanged.
